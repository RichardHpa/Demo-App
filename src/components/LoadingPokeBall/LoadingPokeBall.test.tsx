import { screen, render } from '@testing-library/react'

import { LoadingPokeBall } from './LoadingPokeBall'

describe('LoadingPokeBall', () => {
  it('should render', () => {
    render(<LoadingPokeBall size='100' loading={true} />)
    const pokeBall = screen.getByRole('progress')
    expect(pokeBall).toBeInTheDocument()
    expect(pokeBall).toHaveAttribute('aria-label', 'fetching')
  })

  it('should render with loading false', () => {
    render(<LoadingPokeBall size='100' loading={false} />)
    const pokeBall = screen.getByRole('progress')
    expect(pokeBall).toBeInTheDocument()
    expect(pokeBall).toHaveAttribute('aria-label', 'opened')
  })

  it('should render with opened true', () => {
    render(<LoadingPokeBall size='100' opened={true} />)
    const pokeBall = screen.getByRole('progress')
    expect(pokeBall).toBeInTheDocument()
    expect(pokeBall).toHaveAttribute('aria-label', 'opened')
  })

  it('should render with opened false', () => {
    render(<LoadingPokeBall size='100' opened={false} />)
    const pokeBall = screen.getByRole('progress')
    expect(pokeBall).toBeInTheDocument()
    expect(pokeBall).toHaveAttribute('aria-label', 'fetching')
  })

  it('should render with size 50', () => {
    render(<LoadingPokeBall />)
    const pokeBall = screen.getByRole('progress')
    expect(pokeBall).toBeInTheDocument()
    expect(pokeBall).toHaveAttribute('width', '50')
    expect(pokeBall).toHaveAttribute('height', '50')
  })

  it.each([10, 100])('should render with size %i', (size) => {
    render(<LoadingPokeBall size={size} />)
    const pokeBall = screen.getByRole('progress')
    expect(pokeBall).toBeInTheDocument()
    expect(pokeBall).toHaveAttribute('width', `${size}`)
    expect(pokeBall).toHaveAttribute('height', `${size}`)
  })

  it('should match snapshot', () => {
    const { container } = render(<LoadingPokeBall />)
    expect(container).toMatchSnapshot()
  })

  it.each([10, 100])('should render with size %i', (size) => {
    const { container } = render(<LoadingPokeBall size={size} />)
    expect(container).toMatchSnapshot()
  })
})
