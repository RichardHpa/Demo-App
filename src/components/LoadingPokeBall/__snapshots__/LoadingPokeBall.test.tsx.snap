// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingPokeBall should match snapshot 1`] = `
<div>
  <svg
    aria-label="fetching"
    class="fetching"
    height="50"
    role="progress"
    viewBox="0 0 100 100"
    width="50"
  >
    <g
      transform="translate(50 50) scale(0.8)"
    >
      <g
        transform="translate(0 50)"
      >
        <g
          class="gravity"
        >
          <g
            transform="translate(0 -50)"
          >
            <g
              class="ball"
              transform="scale(1 1)"
            >
              <g
                class="bottom"
              >
                <path
                  d="M -47.5 0 a 47.5 47.5 0 0 0 95 0z"
                  fill="#ffffff"
                  stroke="#303030"
                  stroke-width="5"
                />
              </g>
              <g
                class="top"
              >
                <path
                  d="M -47.5 0 a 47.5 47.5 0 0 1 95 0"
                  fill="#f15d5f"
                />
                <path
                  d="M -38 -0 a 38 38 0 0 1 76 0"
                  fill="none"
                  stroke="#ffffff"
                  stroke-dasharray="0 15 9 9 20 100"
                  stroke-linecap="round"
                  stroke-width="5"
                />
                <path
                  d="M -47.5 0 a 47.5 47.5 0 0 1 95 0z"
                  fill="none"
                  stroke="#303030"
                  stroke-width="5"
                />
              </g>
              <g
                class="open"
                transform="scale(1 0)"
              >
                <path
                  d="M -47.5 -10 a 190 190 0 0 1 95 0 a 190 190 0 0 1 -95 0"
                  fill="#303030"
                  stroke="#303030"
                  stroke-linejoin="round"
                  stroke-width="5"
                />
                <path
                  d="M -47.5 5 a 160 160 0 0 0 95 0 a 180 180 0 0 0 -95 0"
                  fill="#303030"
                  stroke="#303030"
                  stroke-linejoin="round"
                  stroke-width="5"
                />
              </g>
              <g
                class="center"
              >
                <circle
                  cx="0"
                  cy="0"
                  fill="#ffffff"
                  r="12"
                  stroke="#303030"
                  stroke-width="5"
                />
                <circle
                  cx="0"
                  cy="0"
                  fill="#ffffff"
                  r="6"
                  stroke="#303030"
                  stroke-width="3"
                />
                <g
                  class="inner"
                  opacity="0"
                >
                  <circle
                    cx="0"
                    cy="0"
                    fill="#f15d5f"
                    r="4.5"
                  />
                </g>
              </g>
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
</div>
`;

exports[`LoadingPokeBall should render with size 10 1`] = `
<div>
  <svg
    aria-label="fetching"
    class="fetching"
    height="10"
    role="progress"
    viewBox="0 0 100 100"
    width="10"
  >
    <g
      transform="translate(50 50) scale(0.8)"
    >
      <g
        transform="translate(0 50)"
      >
        <g
          class="gravity"
        >
          <g
            transform="translate(0 -50)"
          >
            <g
              class="ball"
              transform="scale(1 1)"
            >
              <g
                class="bottom"
              >
                <path
                  d="M -47.5 0 a 47.5 47.5 0 0 0 95 0z"
                  fill="#ffffff"
                  stroke="#303030"
                  stroke-width="5"
                />
              </g>
              <g
                class="top"
              >
                <path
                  d="M -47.5 0 a 47.5 47.5 0 0 1 95 0"
                  fill="#f15d5f"
                />
                <path
                  d="M -38 -0 a 38 38 0 0 1 76 0"
                  fill="none"
                  stroke="#ffffff"
                  stroke-dasharray="0 15 9 9 20 100"
                  stroke-linecap="round"
                  stroke-width="5"
                />
                <path
                  d="M -47.5 0 a 47.5 47.5 0 0 1 95 0z"
                  fill="none"
                  stroke="#303030"
                  stroke-width="5"
                />
              </g>
              <g
                class="open"
                transform="scale(1 0)"
              >
                <path
                  d="M -47.5 -10 a 190 190 0 0 1 95 0 a 190 190 0 0 1 -95 0"
                  fill="#303030"
                  stroke="#303030"
                  stroke-linejoin="round"
                  stroke-width="5"
                />
                <path
                  d="M -47.5 5 a 160 160 0 0 0 95 0 a 180 180 0 0 0 -95 0"
                  fill="#303030"
                  stroke="#303030"
                  stroke-linejoin="round"
                  stroke-width="5"
                />
              </g>
              <g
                class="center"
              >
                <circle
                  cx="0"
                  cy="0"
                  fill="#ffffff"
                  r="12"
                  stroke="#303030"
                  stroke-width="5"
                />
                <circle
                  cx="0"
                  cy="0"
                  fill="#ffffff"
                  r="6"
                  stroke="#303030"
                  stroke-width="3"
                />
                <g
                  class="inner"
                  opacity="0"
                >
                  <circle
                    cx="0"
                    cy="0"
                    fill="#f15d5f"
                    r="4.5"
                  />
                </g>
              </g>
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
</div>
`;

exports[`LoadingPokeBall should render with size 100 1`] = `
<div>
  <svg
    aria-label="fetching"
    class="fetching"
    height="100"
    role="progress"
    viewBox="0 0 100 100"
    width="100"
  >
    <g
      transform="translate(50 50) scale(0.8)"
    >
      <g
        transform="translate(0 50)"
      >
        <g
          class="gravity"
        >
          <g
            transform="translate(0 -50)"
          >
            <g
              class="ball"
              transform="scale(1 1)"
            >
              <g
                class="bottom"
              >
                <path
                  d="M -47.5 0 a 47.5 47.5 0 0 0 95 0z"
                  fill="#ffffff"
                  stroke="#303030"
                  stroke-width="5"
                />
              </g>
              <g
                class="top"
              >
                <path
                  d="M -47.5 0 a 47.5 47.5 0 0 1 95 0"
                  fill="#f15d5f"
                />
                <path
                  d="M -38 -0 a 38 38 0 0 1 76 0"
                  fill="none"
                  stroke="#ffffff"
                  stroke-dasharray="0 15 9 9 20 100"
                  stroke-linecap="round"
                  stroke-width="5"
                />
                <path
                  d="M -47.5 0 a 47.5 47.5 0 0 1 95 0z"
                  fill="none"
                  stroke="#303030"
                  stroke-width="5"
                />
              </g>
              <g
                class="open"
                transform="scale(1 0)"
              >
                <path
                  d="M -47.5 -10 a 190 190 0 0 1 95 0 a 190 190 0 0 1 -95 0"
                  fill="#303030"
                  stroke="#303030"
                  stroke-linejoin="round"
                  stroke-width="5"
                />
                <path
                  d="M -47.5 5 a 160 160 0 0 0 95 0 a 180 180 0 0 0 -95 0"
                  fill="#303030"
                  stroke="#303030"
                  stroke-linejoin="round"
                  stroke-width="5"
                />
              </g>
              <g
                class="center"
              >
                <circle
                  cx="0"
                  cy="0"
                  fill="#ffffff"
                  r="12"
                  stroke="#303030"
                  stroke-width="5"
                />
                <circle
                  cx="0"
                  cy="0"
                  fill="#ffffff"
                  r="6"
                  stroke="#303030"
                  stroke-width="3"
                />
                <g
                  class="inner"
                  opacity="0"
                >
                  <circle
                    cx="0"
                    cy="0"
                    fill="#f15d5f"
                    r="4.5"
                  />
                </g>
              </g>
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
</div>
`;
